scripts/gluster-block/setup-gluster-block-glusto.yml

```diff
@@ -0,0 +1,392 @@
+    loop_control:
+      pause: 2
+    ignore_errors: True
+
+  - name: Create volume groups
+    lvg:
+      vg: "{{item.0}}"
+      pvs: "/dev/{{item.1}}"
+    ignore_errors: yes
+    with_together:
+    - ['group0', 'group1', 'group2', 'group3']
+    - ['loop0', 'loop1', 'loop2', 'loop3']
+
+  - name: Create thin pool
+    command: "lvcreate -L 45G -T {{item}}/pool"
+    args:
+      creates: "/dev/{{item}}"
+    ignore_errors: yes
+    with_items:
+    - group0
+    - group1
+    - group2
+    - group3
+
+  - name: Create thin volume
+    command: "lvcreate -V 45G -T {{item.0}}/pool -n {{item.1}}"
+    args:
+      creates: "/dev/{{item.0}}/{{item.1}}"
+    ignore_errors: yes
+    with_together:
+    - ['group0', 'group1', 'group2', 'group3']
+    - ['vol0', 'vol1', 'vol2', 'vol3']
+
+  - name: Format the volumes
+    command: "mkfs.xfs /dev/{{item.0}}/{{item.1}}"
+    ignore_errors: yes
+    with_together:
+    - ['group0', 'group1', 'group2', 'group3']
+    - ['vol0', 'vol1', 'vol2', 'vol3']
+
+  - name: Mount the bricks
+    mount: name="/mnt/{{item.0}}" src="/dev/{{item.1}}" state=mounted fstype=xfs
+    with_together:
+    - ['vol0', 'vol1', 'vol2', 'vol3']
+    - ['group0/vol0', 'group1/vol1', 'group2/vol2', 'group3/vol3']
+
+  - name: Start Gluster Daemon
+    service: name=glusterd state=started
+
+  - name: Status rpcbind
+    service: name=rpcbind state=started
+
+
+#Set up gluster-block and related packages on servers
+- hosts: gluster_nodes[1:4]
+  tasks:
+  - name: Install dependency packages
+    yum: name={{ item }} state=installed
+    with_items:
+    - git
+    - libtool
+    - libuuid-devel
+    - python-gobject
+    - glusterfs-api-devel
+    - automake
+    - autoconf
+    - gcc
+    - glib2-devel
+    - "*kmod*"
+    - libnl3*
+    - zlib-devel
+    - cmake
+
+  - name: Clone configshell-fb repo
+    git:
+        repo: https://github.com/open-iscsi/configshell-fb
+        dest: /root/configshell-fb
+        refspec: "+refs/heads/master:refs/remotes/origin/master"
+        version: FETCH_HEAD
+
+  - name: Install rtslib
+    command: "python setup.py install chdir=/root/configshell-fb"
+
+  - name: Clone rtslib repo
+    git:
+        repo: https://github.com/open-iscsi/rtslib-fb
+        dest: /root/rtslib-fb
+        refspec: "+refs/heads/master:refs/remotes/origin/master"
+        version: FETCH_HEAD
+
+  - name: Install rtslib
+    command: "python setup.py install chdir=/root/rtslib-fb"
+
+  - name: Clone targetcli-fb repo
+    git:
+        repo: https://github.com/open-iscsi/targetcli-fb
+        dest: /root/targetcli-fb
+        refspec: "+refs/heads/master:refs/remotes/origin/master"
+        version: FETCH_HEAD
+
+  - name: Install targetcli-fb
+    command: "python setup.py install chdir=/root/targetcli-fb"
+
+  - name: Clone tcmu-runner repo
+    git:
+        repo: https://github.com/open-iscsi/tcmu-runner
+        dest: /root/tcmu-runner
+        refspec: "+refs/heads/master:refs/remotes/origin/master"
```

Comment:
If you're cloning from github and just want master, you can most likely avoid refspec